### PR TITLE
Merge End2End and integration tests

### PR DESCRIPTION
**Pull request by velezd**
There is now only small difference between end2end and standard
integration style tests, so it doesn't make sense to maintain both of
them separately.

### CODE REVIEW

**Review by VladimirSlavik — COMMENTED** [review]
I'm not sure what is the intention with commits - I'd expect one large commit or multiple smaller ones? Other than that + comments below, mostly LGTM.

File: test/check-progress [hunk 1]
```diff
@@ -55,10 +54,7 @@ class TestInstallationProgress(anacondalib.VirtInstallMachineCase):
             ignore=["#betanag-icon"],
         )
 
-        p.wait_done()
-        add_public_key(self.machine)
-        p.reboot()
-        m.wait_reboot()
+        self.reboot()
```
[the remark this answers is not in the document]
Comment (velezd) [reply test/check-progress:57]: - Adding the public key is now part of the reboot, maybe I should change name of the function so it's not confusing.
- I just moved `wait_done` up, it was used incorrectly.
- I will split this into separate commit.

File: test/anacondalib.py [hunk 2]
```diff
@@ -70,3 +76,34 @@ def resetStorage(self):
...
+        self.machine.download('/tmp/packaging.log', 'packaging.log', self.logs_dir)
+        self.machine.download('/tmp/storage.log', 'storage.log', self.logs_dir)
+        self.machine.download('/tmp/dbus.log', 'dbus.log', self.logs_dir)
+        self.machine.download('/tmp/syslog', 'syslog', self.logs_dir)
+        try:
+            self.machine.download('/tmp/anaconda-tb-*', '.', self.logs_dir)
+        except CalledProcessError:
+            pass
+
+    def reboot(self):
```
Comment (KKoukiou) [inline test/anacondalib.py:98]: It's slightly confusing to have two definitions of the reboot method. I think this code should be better incorporated with the reboot method in the test/helpers/progress.

Comment (velezd) [reply test/anacondalib.py]: I've renamed the method to make it less confusing. It does more than what Progress helper should handle.

File: test/check-storage [hunk 3]
```diff
@@ -1052,5 +1054,51 @@ class TestStorageMountPointsEFI(anacondalib.VirtInstallMachineCase):
...
+
+    def testAfterReboot(self):
+        i = Installer(self.b, self.m)
+        pas = Password(self.b, self.s.encryption_id_prefix)
+        pro = Progress(self.b)
+
+        i.open()
+        i.reach(i.steps.DISK_ENCRYPTION)
+
+        # Configure storage encryption
```
[the remark this answers is not in the document]
Comment (velezd) [reply test/check-storage:1073]: Or I could just add the post-reboot check to the testAutopartitioning / testAutopartitioningEncrypted test.

[the remark this answers is not in the document]
Comment (KKoukiou) [reply test/check-storage:1073]: That would be also ok, yes!

**Review by KKoukiou — APPROVED** [review]
Assuming CI is green.

**Review by KKoukiou — CHANGES_REQUESTED** [review]
Needs more test care.

**Comment by rvykydal** [comment]
The error from the failing tests: https://cockpit-logs.us-east-1.linodeobjects.com/pull-64-20231212-115533-08a52cf2-fedora-rawhide-boot/log.html:
```
Traceback (most recent call last):
  File "/work/bots/make-checkout-workdir/test/anacondalib.py", line 115, in tearDown
    super().tearDown()
  File "/work/bots/make-checkout-workdir/test/common/testlib.py", line 1613, in tearDown
    self.check_browser_errors()
  File "/work/bots/make-checkout-workdir/test/common/testlib.py", line 1926, in check_browser_errors
    raise Error(UNEXPECTED_MESSAGE + "browser errors:\n" + log)
testlib.Error: FAIL: Test completed, but found unexpected browser errors:
error: Localization client closed
```
seems somehow familiar to me. I'll try to remember what was it.

**Comment by rvykydal** [comment]
> The error from the failing tests: https://cockpit-logs.us-east-1.linodeobjects.com/pull-64-20231212-115533-08a52cf2-fedora-rawhide-boot/log.html:
> 
> ```
> Traceback (most recent call last):
>   File "/work/bots/make-checkout-workdir/test/anacondalib.py", line 115, in tearDown
>     super().tearDown()
>   File "/work/bots/make-checkout-workdir/test/common/testlib.py", line 1613, in tearDown
>     self.check_browser_errors()
>   File "/work/bots/make-checkout-workdir/test/common/testlib.py", line 1926, in check_browser_errors
>     raise Error(UNEXPECTED_MESSAGE + "browser errors:\n" + log)
> testlib.Error: FAIL: Test completed, but found unexpected browser errors:
> error: Localization client closed
> ```
> 
> seems somehow familiar to me. I'll try to remember what was it.

... ah, maybe it was addCleanup handlers run for destructive tests, so perhaps not this case

**Review by KKoukiou — APPROVED** [review]
As long as CI is green

**Pull request merged**